client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/allocation/utils.ts

```diff
@@ -0,0 +1,73 @@
+import { NumUtils } from '@common/utils';
+import { DraftStockOutLineFragment } from '../../../api/operations.generated';
+import { DateUtils } from '@common/intl';
+
+export const sumAvailableQuantity = (
+  draftLines: DraftStockOutLineFragment[]
+) => {
+  const sum = draftLines.reduce(
+    (acc, { stockLineOnHold, availablePacks, packSize, location }) =>
+      !location?.onHold && !stockLineOnHold
+        ? acc + availablePacks * packSize
+        : acc,
+    0
+  );
+
+  return sum;
+};
+
+export const getAllocatedUnits = ({
+  draftLines,
+  placeholderQuantity,
+}: {
+  draftLines: DraftStockOutLineFragment[];
+  placeholderQuantity: number | null;
+}) =>
+  NumUtils.round(
+    draftLines.reduce(
+      (acc, { numberOfPacks, packSize }) => acc + numberOfPacks * packSize,
+      0
+    ),
+    3
+  ) + (placeholderQuantity ?? 0);
```

Comment:
This is another one to add to description of behaviour, if we have placeholder less then what you can supply, and we manually supply that, placeholder stays. We kind of need something similar to prescription, 'prescribed' quantity or it could be placeholder, but the rule is, placeholder is populated with how much you want to issue, but is displayed as placholder quantity - how much you've issued